CHANGELOG.md

```diff
@@ -1,7 +1,75 @@
 Change Log
 ==========
 
-# Version v4.0.0
+# Version 4.0.1
+
+_2024-10-01_
+
+This release contains a handful of bug fixes and improvements. 
+
+## 🧩 Compiler Plugins additions (experimental)
+
+The [compiler plugins](https://www.apollographql.com/docs/kotlin/advanced/compiler-plugins) API has been extended to allow listening to schema changes.\
+This can be used to have plugins generate code based on the schema. To do this, implement the [`schemaListener`](https://www.apollographql.com/docs/kotlin/kdoc/apollo-compiler/com.apollographql.apollo.compiler/-apollo-compiler-plugin/schema-listener.html)
+function on your plugin:
+    
+```kotlin
+class MyCompilerPlugin() : ApolloCompilerPlugin {
+  @ApolloExperimental
+  override fun schemaListener(): SchemaListener {
+    return object : SchemaListener {
+      override fun onSchema(schema: Schema, outputDirectory: File) {
+        // Generate some code inside outputDirectory based on schema
+      }
+    }
+  }
+}
+```
```

Comment:
If we're highlighting `ApolloCompilerPlugin.schemaListener`, we should also highlight generic support for scalars.

Reply:
I was hesitating but feel like it's more or less a bug fix 😅.